portfolio/src/main/webapp/slideshow_script.js

```diff
@@ -0,0 +1,55 @@
+//sliding
+var slideIndex = 1;
+var timeout;
+showSlide(slideIndex); //initial state
+
+function showSlide(index){
+    var slides = document.getElementsByClassName("slide");
+    var noSlides = slides.length
+    console.log(noSlides);
+    if(index>noSlides || index<1 ){
+        index=(index%noSlides+noSlides)%noSlides+1;
+    }
+    for(var i=0; i<noSlides; i++){
+        slides[i].style.display="none";
```

Comment:
Another micro-nit and just personal opinion: it could be debatable if we want the function `showSlide(index)` to also hide the other slides.  Imagine you want in the future to have a gallery where you show multiple slides on the same screen. We would have to change this function quite a bit. Ideally we have multiple small re-usable functions. 

We could probably create a few helper function for this `hideSlide(index)` and `hideAllSlides()` and then call these in `plusSlides()`.

Then the way flow could look like:

```
function plusSlides(n) {
  slideIndex = incrementSlideIndexBy(n);
  hideAllSlides();
  showSlide(slideIndex);
}
```

And then in the future if we want to show multiple slides per page, we could change the function `plusSlides(n)` and the other functions could remain unchanged.

Reply:
I restructured these functions, extracted a hideObject() and showObject() function and built both the sliding and the button+captions hiding functionality on top of these.